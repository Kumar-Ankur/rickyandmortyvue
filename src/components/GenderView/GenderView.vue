<template>
  <div class="species-content">
    <h2>{{ GENDER_FILTER }}</h2>

    <app-check-box
      v-for="(data, index) in genderData"
      :key="index"
      :data="data"
      :addFilter="addFilter"
      :removeFilter="removeFilter"
      :modelName="modelName"
      :store="store"
    ></app-check-box>
  </div>
</template>

<script>
import CheckBox from "../CheckBox/CheckBox.vue";
import * as constants from "../../constants";
import { getCheckBoxData } from "../../utils";
export default {
  data() {
    return {
      GENDER: constants.GENDER,
      GENDER_FILTER: constants.GENDER_FILTER,
      genderData: [],
      modelName: 'gender',
      addFilter: this.$store.dispatch("addGenderFilteredData"),
      removeFilter: this.$store.dispatch("removeGenderFilteredData"),
      store: this.$store
    };
  },
  components: {
    appCheckBox: CheckBox
  },
  computed: {
    characters() {
      return getCheckBoxData(this.$store.getters.characters, constants.GENDER);
    }
  },
  watch: {
    characters(newVal, oldVal) {
      this.genderData = newVal;
    }
  }
};
</script>

<style scoped>
.species-content {
  color: white;
  border: 1px solid white;
  padding: 2rem;
  margin-left: 2rem;
}
</style>