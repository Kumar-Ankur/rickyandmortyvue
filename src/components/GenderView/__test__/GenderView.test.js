import { shallowMount } from '@vue/test-utils';
import GenderView from '../GenderView.vue';

test('generate snapshot', () => {
    const wrapper = shallowMount(GenderView, {
        mocks: {
            $store: {
                getters: {
                    characters: []
                }
            }
        }
    })
    expect(wrapper.html()).toMatchSnapshot();
})