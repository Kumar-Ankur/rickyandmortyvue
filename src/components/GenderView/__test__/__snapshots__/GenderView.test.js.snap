// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate snapshot 1`] = `
"<div class=\\"species-content\\">
  <h2>Gender Filter:</h2>
</div>"
`;
